src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -19,6 +19,7 @@ internal sealed class DynamicAnalysisInjector : CompoundInstrumenter
         private readonly MethodSymbol _method;
         private readonly BoundStatement _methodBody;
         private readonly MethodSymbol _createPayload;
+        private readonly PooledHashSet<DebugSourceDocument> _documentsBuilder;
```

Comment:
With either approach, we need to check the document at each span. The only difference is where the check is done, in `AddAnalysisPoint` or in `CreateBlockPrologue`. I'd prefer one less field.